src/components/Auth/SignupOption.tsx

```diff
@@ -1,31 +1,70 @@
+import { usePostEmailMutation, usePostNicknameMutation } from 'features/auth/authApi';
 import React from 'react';
 import styled from 'styled-components';
 import { theme } from 'styles/theme';
+import { UserSignupRequest } from 'types/auth';
 
 interface SignupOptionProps {
   type: string;
   error: boolean;
+  value: string;
+  isCompleteList: { [key: string]: boolean };
+  handleConfirm: (type: keyof UserSignupRequest, value: boolean) => void;
 }
 function SignupOption(props: SignupOptionProps) {
-  const { type, error } = props;
+  const { type, error, value, handleConfirm, isCompleteList } = props;
+  const [checkNickname] = usePostNicknameMutation();
+  const [checkEmail] = usePostEmailMutation();
+  const handleClick = async () => {
+    if (type === 'nickname') {
+      try {
+        const { uniqueNickname } = await checkNickname({ nickname: value }).unwrap();
+        handleConfirm(type, uniqueNickname);
+      } catch (error) {
+        console.log(error);
+      }
+    } else if (type === 'email') {
+      try {
+        const { uniqueEmail, verificationCode } = await checkEmail({ email: value }).unwrap();
+        handleConfirm(type, uniqueEmail);
+        localStorage.setItem('verificationCode', verificationCode);
+      } catch (error) {
+        console.log(error);
+      }
+    } else if (type === 'emailConfirm') {
+      const verificationCode = localStorage.getItem('verificationCode');
+      handleConfirm(type, value === verificationCode);
+      localStorage.removeItem('verificationCode');
+    }
+  };
 
   const getSignOption = (type: string) => {
```

Comment:
```suggestion
const getSignOption = (type: string) => {
    switch (type) {
      case 'email':
        return {
          disabled: error,
          label: '인증번호 전송',
        };
      case 'nickname':
        return {
          disabled: error || isCompleteList.nickname,
          label: '중복확인',
        };
      case 'emailConfirm':
        return {
          disabled: error || isCompleteList.emailConfirm,
          label: '확인',
        };
      default:
        return null;
    }
  };

  const currentOption = getSignOption(type);

  if (!currentOption) return null;
  const { disabled, label } = currentOption;

  return (
    <StyledBtn onClick={handleClick} tabIndex={-1} inputType={type} disabled={disabled}>
      {label}
    </StyledBtn>
  );
```

이런 방식도 있을 것 같아요!
switch문으로 분기하고 싶은 부분이 disabled랑 텍스트 부분밖에 없는 것 같아서용